### PR TITLE
Filtering, OAuth, Bootstrap, Projects Public Page

### PR DESCRIPTION
**Pull request by AlexanderHallCS**
These are large changes from a while ago. Lots of files added, changed.

### CODE REVIEW

**Comment by atikanc** [comment]
The time is now! Will start reviewing these changes now.

**Review by tyler642101 — APPROVED** [review]
This is it. The greatest pull request to ever live.

**Comment by cjnewman820** [comment]
Looks great, making great progress!

**Review by cjnewman820 — APPROVED** [review]

File: app/views/projects/public.html.erb [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        </div>
+        
+        <div class="col">
+<<<<<<< HEAD
+            <div class="card h-100">
+                <button type="button" class="btn btn-default" data-toggle="modal" data-target="#form-modal-<%= project.id %>" id="<%= project.ProjectName %>">
+=======
+            <div class="card h-100 grey-background-ray blue-outline-ray">
+                <button type="button" class="btn btn-default" data-toggle="modal" data-target="#form-modal-<%= project.id %>">
+>>>>>>> c720c76d9cc4d6d9015905ee5a8f798e9bba1187
```
Comment (atikanc) [inline app/views/projects/public.html.erb:114]: Some merge conflict text needs to be removed.

File: spec/models/admin_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Admin, type: :model do
+  subject do
+    described_class.new(email: 'mockemail@email.com', full_name: 'Test Name', uid: '123456789', avatar_url: 'https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png')
```
Comment (atikanc) [inline spec/models/admin_spec.rb:5]: Should this be the roboburner email?

**Review by atikanc — CHANGES_REQUESTED** [review]
Change request mentioned in conversation.

**Review by atikanc — APPROVED** [review]
Looks great! Thanks for making the changes.

**Review by Bao-Vo-Creator — COMMENTED** [review]
Everything is good.

**Pull request merged**